### ISSUE TITLE
DatePicker chinese locale overflow

### ISSUE THREAD

**Issue by lishuhao** [flutter/flutter#29736]
## When set chinese locale ,datepicker overflowed
![](https://raw.githubusercontent.com/lishuhao/assets/master/issues/WechatIMG523.jpeg)

```
👉  issues git:(master) ✗ flutter doctor -v
[✓] Flutter (Channel beta, v1.3.8, on Mac OS X 10.14.3 18D109, locale zh-Hans-CN)
    • Flutter version 1.3.8 at /Users/hao/Documents/flutter
    • Framework revision e5b1ed7a7f (2 weeks ago), 2019-03-06 14:23:37 -0800
    • Engine revision f4951df193
    • Dart version 2.2.1 (build 2.2.1-dev.0.0 571ea80e11)

[!] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /Users/hao/Library/Android/sdk
    • Android NDK at /Users/hao/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = /Users/hao/Library/Android/sdk
    • Java binary at: /Applications/Android Studio 2.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    ! Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[✓] iOS toolchain - develop for iOS devices (Xcode 10.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.1, Build version 10B61
    • ios-deploy 1.9.4
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.3)
    • Android Studio at /Applications/Android Studio 2.app/Contents
    • Flutter plugin version 33.4.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] VS Code (version 1.32.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 2.24.0

[✓] Connected device (2 available)
    • ONE E1001 • bb72222f                             • android-arm • Android 6.0.1 (API 23)
    • iPhone 6s • 0F7B433C-9767-491B-A929-2663B1C9F87A • ios         • iOS 12.1 (simulator)

! Doctor found issues in 1 category.
```

**Comment by karmactonics** [comment]
I have same probrem with Japanese Language.
![IMG_E0027](https://user-images.githubusercontent.com/8557013/54795318-06d27180-4c8f-11e9-9d66-c6983f9c4a3d.JPG)

**Comment by darrenaustin** [comment]
This should be fixed with #31514.
